### PR TITLE
Get Tutum credentials from S3

### PR DESCRIPTION
**Pull request by jskeates**
### Problem

#2

### Solution

One potential way to solve #2. Read in from S3 using the AWS CLI, with the EC2 Instance's IAM Role acting as the authentication.

For this to work, you need to set up an S3 bucket and an IAM Instance Role with permission to read from that bucket. You then need to create a folder in the bucket with the name of your environment (e.g. `staging`). In the folder, put a plain text file with your username and then another with your API key. So, from your local machine:
```bash
echo "jskeates" | aws s3 cp - s3://my-tutum-bucket/staging/user
echo "123456abc..." | aws s3 cp - s3://my-tutum-bucket/staging/api_key
```
You should then change the `CREDENTIALS_S3_BUCKET` and `ENVIRONMENT` variables in the script under the "Parameters" section appropriately.
```bash
CREDENTIALS_S3_BUCKET="my-tutum-bucket"
ENVIRONMENT="staging"
```

This also changes where we install the AWS CLI from, as the version in Ubuntu's repos is too old to work properly.

Any thoughts/feedback?

### CODE REVIEW

File: script.sh [hunk 1]
```diff
@@ -1,6 +1,14 @@
 #!/bin/bash
 # Expected to be run on Ubuntu 14.04
 
+# Parameters
+# -- Tutum Credentials: store your username and API Key in an S3 bucket that the
+# EC2 Instance's IAM Role has permission to access
+# Username: <BucketRoot>/<Environment>/user
```
Comment (revett) [inline script.sh]: Should be:

```
<CREDENTIALS_S3_BUCKET>/<Environment>/user
```

And same for line 8.

Comment (revett) [inline script.sh]: Can the file also be prefixed with `tutum_auth_x`?

File: script.sh [hunk 2]
```diff
@@ -17,17 +25,17 @@ sudo apt-get install jq -y
 
 # Install AWS CLI
 
-sudo apt-get install awscli -y
-
-# Set Tutum env vars
-
-export TUTUM_USER=xxx
-export TUTUM_APIKEY=xxx
+sudo pip install awscli
```
Comment (revett) [inline script.sh]: Install a specific version of `awscli`?

**Comment by revett** [comment]
@jskeates added some comments - thanks for the 2nd PR!

**Comment by jskeates** [comment]
Updated from your comments. Pinned the CLI to the latest version 1.9.20.

**Comment by revett** [comment]
:cake:

**Pull request closed without merging**